addons/vagrant/playbooks/linux_servers/samba4ad.yml

```diff
@@ -9,190 +9,217 @@
-    
-  - name: add apt source
-    ansible.builtin.apt_repository:
-      repo: "deb https://samba.tranquil.it/debian/samba-4.16/ bullseye main"
-      state: present
-
-  - name: update and upgrade
-    ansible.builtin.apt:
-      upgrade: yes
-      update_cache: yes
-
-  - name: Install Samba packages
-    ansible.builtin.package:
-      name: '{{ item }}'
-      state: 'present'
-    loop: '{{ samba4ad__base_packages }}'
-
-  - name: Stop and Mask Services
-    ansible.builtin.systemd:
-      enabled: no
-      state: stopped
-      masked:  yes
-      name: "{{ item }}"
-    loop: "{{ samba4ad__services_to_stop }}"
-
-  - name: Remove config files
-    file:
-      path: "{{ item }}"
-      state: absent
-    loop:
-      - "/etc/samba/smb.conf"
-      - "/etc/krb5.conf"
-
-  - name: change krb5.conf
-    ansible.builtin.blockinfile:
-      create: true
-      dest: /etc/krb5.conf
-      content: |
-         [libdefaults]
-            default_realm = "{{ dns_domain_upper }}"
-            dns_lookup_kdc = "{{ samba4ad__kdc }}"
-            dns_lookup_realm = "{{ samba4ad__realm }}"
-         
-         [realms]
-            "{{ dns_domain_upper }}" = {
-              "admin_server = AD.{{ dns_domain_upper }}"
-              "kdc = AD.{{ dns_domain_upper }}"
-         }
-         
-         [domain_realm]
-           "{{ dns_domain }} = {{ dns_domain_upper }}"
-           ".{{ dns_domain }} = {{ dns_domain_upper }}"
-
-  # to have persistant dns serevrs in resolve.conf 
-  - name: Adding localhost in resolv.conf
-    ansible.builtin.lineinfile:
-      path: /etc/dhcp/dhclient.conf
-      line: prepend domain-name-servers 127.0.0.1, 8.8.8.8;
-      state: present
-      create: true
-
-  - name: Configure Samba
-    ansible.builtin.shell:
-      cmd: "{{ item }}"
-    loop:
-    - "samba-tool domain provision --realm={{ dns_domain_upper }} --domain {{ samba4ad__global_custom['workgroup'] }} --server-role=dc"
-
-  - name: Allow ntml auth in samba config
+  - name: Check if samba4ad configuration is there
     ansible.builtin.lineinfile:
       path: /etc/samba/smb.conf
-      regexp: '^\s*ntlm auth = yes'
-      insertafter: '^\s*workgroup.*'
       line: "        ntlm auth = yes"
-
-  - name: Configure Samba admin password
-    ansible.builtin.expect:
-      command: "{{ item }}"
-      responses:
-        (?i)new\s(?i)password: "{{ samba4ad__password }}"
-        (?i)retype\s(?i)password: "{{ samba4ad__password }}"
-    loop:
-    - "samba-tool user setpassword {{ samba4ad__username }}"
-
-  - name: Remove config files
-    file:
-      path: "{{ item }}"
-      state: absent
-    loop:
-    - "/var/lib/samba/private/krb5.conf"
-
-  - name: Creating a symlink
-    ansible.builtin.file:
-      src: "/etc/krb5.conf"
-      dest: "/var/lib/samba/private/krb5.conf"
-      state: link
-
-  - name: Unmask and start Services
-    ansible.builtin.systemd:
-      enabled: yes
+    check_mode: yes
+    register: check_if_samba
+    ignore_errors: true
```

Comment:
I don't think you need "ignore_errors"

Reply:
Unfortunately, It does not work without. I think it is due to the ansible version that we use.